Sources/NIO/PipePair.swift

```diff
@@ -0,0 +1,139 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension NIOFileHandle: Selectable {
+}
+
+final class PipePair: SocketProtocol {
+    typealias SocketType = NIOFileHandle
+
+    let inputFD: NIOFileHandle
+    let outputFD: NIOFileHandle
+
+    init(inputFD: NIOFileHandle, outputFD: NIOFileHandle) {
+        self.inputFD = inputFD
+        self.outputFD = outputFD
+        try! inputFD.withUnsafeFileDescriptor {
```

Comment:
I assume the plan isn't to crash if either of these things fail.

Reply:
thanks, fixing.